pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/SchemaUtils.java

```diff
@@ -44,6 +47,13 @@
   private SchemaUtils() {
   }
 
+  // checker to ensure simple date format matches lexicographic ordering.
+  private static final Map<Character, Integer> DATETIME_PATTERN_ORDERING = new HashMap<>();
+  static {
+    DATETIME_PATTERN_ORDERING.putAll(ImmutableMap.of('y', 0, 'M', 1, 'd', 2));
+    DATETIME_PATTERN_ORDERING.putAll(ImmutableMap.of('H', 3, 'm', 4, 's', 5, 'S', 6));
```

Comment:
(minor) Create an array to instantiate the map for readability
```suggestion
    char[] patternOrdering = new char[]{'y', 'M', 'd', 'H', 'm', 's', 'S'};
    for (int i = 0; i < patternOrdering.length; i++) {
      DATETIME_PATTERN_ORDERING.put(patternOrdering[i], i);
    }
```